source/fe/mapping_q.cc

```diff
@@ -1077,7 +1153,18 @@ MappingQ<dim, spacedim>::fill_fe_subface_values(
        &data.cell_of_current_support_points->get_triangulation()) ||
       (cell != data.cell_of_current_support_points))
     {
-      data.mapping_support_points = this->compute_mapping_support_points(cell);
+      if (polynomial_degree > 1)
+        data.mapping_support_points =
+          this->compute_mapping_support_points(cell);
+      else
+        {
+          data.mapping_support_points.resize(
+            GeometryInfo<dim>::vertices_per_cell);
+          const auto vertices = this->get_vertices(cell);
+          for (unsigned int i = 0; i < GeometryInfo<dim>::vertices_per_cell;
+               ++i)
+            data.mapping_support_points[i] = vertices[i];
```

Comment:
Have you considered changing the type of `mapping_support_points` to `boost::small_vector`? I suspect that would not require large-scale changes, but you would not have to do this hand-copying.

Reply:
Which you do about a dozen times in this patch :-)

Reply:
The problem I have there is that `compute_mapping_support_points` is a protected member function, so it is part of our API. In fact, our derived class `MappingQCache` stores exactly those points https://github.com/dealii/dealii/blob/a8379e2009bdfcb9e6fcb9fdb7dd074668217145/include/deal.II/fe/mapping_q_cache.h#L218-L224 and has functions to fill `std::vector` like https://github.com/dealii/dealii/blob/a8379e2009bdfcb9e6fcb9fdb7dd074668217145/include/deal.II/fe/mapping_q_cache.h#L119-L122 and some related ones. Of course, nothing here is really insurmountable, but I would at least need to think more about it, so I'll leave it to another PR. (Of course, it would make for a nice simplification in the future if we figure out it works well.)

Reply:
Fair point. I think I'm not overly worried about breaking obscure `protected` interfaces. The number of people who have a class derived from `Mapping` is likely countable with a two-digit binary number...